infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespaceFactory.java

```diff
@@ -531,4 +616,11 @@ protected String getServiceAccountName() {
   protected Set<String> getClusterRoleNames() {
     return clusterRoleNames;
   }
+
+  private String crc32(String in) {
```

Comment:
what are the benefits of using crc32 over using the string directly?

Reply:
We never need the actual value; we need only fact it is cnanged (or not). So this is more compact storing form (like timestamps etc). If you need to track file content change, will you store whole file? No..  
Don't  forget it will be stored into each and every user preference. 
Also direct exposing of the template value may have security concerns under some circumstances? (questionable).

Reply:
I don't see security reason of eventual exposing template from this. You're storing it before placeholder evaluation, so it's just copy of che-server configuration value from the time workspace was created.

With compact string, yes it's "probably" shorter. Considering default `<username>-che`, which has crc32 value `3677644488`, we save 4 bytes per user. I can't imagine number of users that would make this number even close to save something significant and didn't cause us issues somewhere else.
When the template is `<username>`, we save exactly nothing. When template is `<userid>`, we even waste 2 bytes.

So memory saving is uncertain, because the source string is typically short. It's limited by kubernetes boundaries, which we know is reasonably short. What we know that it cost CPU time to compute the hash. Both is so negligible that I don't see a reason to calculate hash here.

With tracking file content change, you're right. But we're not comparing files that could have megs or gigs. We know we have string that is <100 characters and complicate stuff for humans reading the code, just to save few bytes, does not make sense here in my opinion.

my vote goes to prefer readability and simplicity here.

Reply:
Templates is not always as short. At least it can be `che-<userid>-<workspaceid>` etc. About exposing, agree it not a big secret, but anyway its kind of showing system-wide setting for end user.

Reply:
>my vote goes to prefer readability and simplicity here.
+1